src/collision/CollisionRules_test.js

```diff
@@ -0,0 +1,55 @@
+import assert from 'assert';
+import {createArgSaver} from '../testing/mocks.js';
+import {CollisionRules} from './CollisionRules.js';
+import {Vec2, Size} from '../core.js';
+import {Enemy, Bullet, PlasmaBullet} from '../entities/all.js';
+
+describe('CollisionRules', () => {
+  describe('onCollide', () => {
+    const enemy = new Enemy(new Vec2(0,0), new Size(0,0), null, 0, 0);
+    
```

Comment:
We also want a test case where the collision function is not called. It's currently possible to pass these tests with an implementation that doesn't even include any `instanceof` checks